lib/manageiq/providers/ibm_cloud/cloud_tools/common.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+dir_name = File.basename(__FILE__).split('.')[0]
+Dir.glob(File.join(dir_name, '*.rb'), :base => File.dirname(__FILE__)) { |req| require_relative req }
```

Comment:
So this all can just be:

```suggestion
Dir.glob(File.join(File.expand_path(__dir__), "*.rb").sort.each { |r| require r }
```

Note the `.sort`, which is handy to ensure consistent ordering, otherwise glob's order is not guaranteed.

Reply:
require_relative is less overhead on the system as it skips a lot of logic path lookup logic. I don't want to get the .rb files in the parent directory I want to get the files in the sub directory.

```
puts 'My implementation'
dir_name = File.basename(__FILE__).split('.')[0]
puts dir_name
Dir.glob(File.join(dir_name, '*.rb'), :base => File.dirname(__FILE__)) { |req| puts "require_relative(#{req})"; require_relative req }

puts "Jason's change"
Dir.glob(File.join(File.expand_path(__dir__), "*.rb")).sort.each { |r| puts "require(#{r})"; require r }
```

```
My implementation
common
require_relative(common/timeout.rb)
require_relative(common/iam_auth.rb)
require_relative(common/proxy.rb)

Jason's change
require(/.../manageiq-providers-ibm_cloud/lib/manageiq/providers/ibm_cloud/cloud_tools/common.rb)
require(/.../manageiq-providers-ibm_cloud/lib/manageiq/providers/ibm_cloud/cloud_tools/sdk.rb)
require(/.../manageiq-providers-ibm_cloud/lib/manageiq/providers/ibm_cloud/cloud_tools/taggings.rb)
require(/.../manageiq-providers-ibm_cloud/lib/manageiq/providers/ibm_cloud/cloud_tools/vpc.rb)
require(/.../manageiq-providers-ibm_cloud/lib/manageiq/providers/ibm_cloud/cloud_tools/vpc_sdk.rb)
```

Reply:
If we're worried about the runtime cost of require vs require_relative we could also speed this up by dropping the `Dir.glob` and just list out the requires:
```
require_relative(common/timeout.rb)
require_relative(common/iam_auth.rb)
require_relative(common/proxy.rb)
```

Reply:
True, however listing out the requires would hinder reorganization. I'm pointing out that my implementation is both performative and flexible.

Reply:
I'm not saying your code is wrong, I'm saying it can be cleaner by using __dir__, which was a mistake on my part because it was hard to read.

That being said, if we care about performance, I believe the fastest is using plain require with absolute paths (which is what I was trying to write, and why I added expand_path), then require_relative then require with other paths.

Reply:
Oh I see what I did...I *meant* to write,

```ruby
Dir.glob(File.join(File.expand_path(__dir__), "common", "*.rb").sort.each { |r| require r }
```

which is the require pattern we use elsewhere in the code base.